### PR TITLE
bump gradle from 7.3.1 to 7.5.1

### PR DESCRIPTION
**Pull request by asolntsev**
## Checklist

 - [ ] Read the [contributing guide](https://github.com/mockito/mockito/blob/main/.github/CONTRIBUTING.md)
 - [ ] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [ ] If possible / relevant include an example in the description, that could help all readers
       including project members to get a better picture of the change
 - [x] Avoid other runtime dependencies
 - [ ] Meaningful commit history ; intention is important please rebase your commit history so that each
       commit is meaningful and help the people that will explore a change in 2 years
 - [x] The pull request follows coding style
 - [ ] Mention `Fixes #<issue number>` in the description _if relevant_
 - [ ] At least one commit should mention `Fixes #<issue number>` _if relevant_

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=9afb3ca688fc12c761a0e9e4321e4d24e977a4a8916c8a768b1fe05ddb4d6b66
```
Comment (TimvdLippe) [inline gradle/wrapper/gradle-wrapper.properties:3]: Please add the 7.5.1 sha here, to make sure we use the valid Gradle wrapper.

**Review by TimvdLippe — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (asolntsev) [reply gradle/wrapper/gradle-wrapper.properties:3]: @TimvdLippe Thanks, I have added `distributionSha256Sum`. It's strange that `gradlew wraper` didn't add it automatically.

P.S. But are you it really works? I tried to put just random number to `distributionSha256Sum` value, and gradle still worked...

Comment (TimvdLippe) [reply gradle/wrapper/gradle-wrapper.properties:3]: Our GitHub action should verify that we are using the correct version

**Review by TimvdLippe — APPROVED** [review]
Thanks!

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (asolntsev) [reply gradle/wrapper/gradle-wrapper.properties:3]: But both builds in github actions were green...
Apparently, the checksum was not verified.

image

Comment (TimvdLippe) [reply gradle/wrapper/gradle-wrapper.properties:3]: This is supposed to be verified by https://github.com/mockito/mockito/blob/eb85518cef362e39788a7e030bab8ac78eca76c9/.github/workflows/ci.yml#L52

Maybe this verification only runs if a hash exists? Or did we misconfigure the action?

Comment (asolntsev) [reply gradle/wrapper/gradle-wrapper.properties:3]: Oh yes, that's right: the verification runs only if checksum exists.
I tried to enter a wrong checksum, and the build failed:
> Expected checksum: 'NOPE596b10cce501591e92f229816aa4046424f3b24d771751b06779d58WRONG'
>  Actual checksum: 'f6b8596b10cce501591e92f229816aa4046424f3b24d771751b06779d58c8ec4'

see https://github.com/mockito/mockito/pull/2778

**Pull request merged**